Added new Rmd template for basic PEcAn workflow to PEcAn.all package

dlebauer opened the pull request:
## Description

Added a template for a basic PEcAn workflow.

## Motivation and Context

Will help users make the transition from Web to interactive R session by adding an Rmarkdown template that users can use to create a new workflow from within the Rstudio interface.

![image](https://user-images.githubusercontent.com/464871/31832337-4939a7e6-b58c-11e7-8d68-12f349a4bd34.png)

I didn't know where to put the documentation, but it would be 

In Rstudio with the PEcAn.all package loaded 

```r
library(PEcAn.all)
```

* File --> new file --> Rmarkdown --> From Template
* Choose the PEcAn workflow template.
* You can start with a pecan.xml file from the web interface or the tests/ directory and modify as needed

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) <!-- please add issue number -->
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

<!--this template is from https://www.talater.com/open-source-templates/#/page/99-->

dlebauer committed:
Added new Rmd template for basic PEcAn workflow to PEcAn.all package

tonygardella commented on base/all/inst/rmarkdown/templates/pecan_workflow/skeleton/skeleton.Rmd:
````diff
@@ -0,0 +1,61 @@
+```{r read.settings}
+settings <- PEcAn.settings::read.settings("pecan.xml")
+```
+
````
Do conversions:
```
settings <- PEcAn.utils::do_conversions(settings)
```
Traits:
```
settings <- PEcAn.DB::runModule.get.trait.data(settings)
PEcAn.settings::write.settings(settings, outputfile='pecan.TRAIT.xml')
```

tonygardella reviewed (changes requested):
Super useful! Need the do.conversions and Traits step.

mdietze reviewed (changes requested):
Let's discuss this at next week's meeting. We had previously worked hard to migrate from having 3 different versions of workflow.R to having having a single canonical version because having multiple workflows was causing a ton of problems both for development (having to implement the same thing in multiple places) and debugging (failing to implement the same thing in multiple places). The fact that @tonygardella already identified such a discrepancy within the originating PR is pretty indicative that the proposed addition will revert us to that condition again.

dlebauer commented:
@mdietze I thought about discussing first but wanted to put this up there as a starting point. 

Perhaps if we really make this bare-bones and/or automatically parse lines from workflow.R during one of the make steps, we could keep them consistent. But I think that having an Rmd template is a fundamentally different use case than having an automated R script. The scripts/workflow*.R files are not user friendly for interactively debuging a workflow, or for keeping track of science methods and rationale , notes etc. and ultimately publishing a workflow as a reproducible(ish?) analysis. The status and if/else get in the way. 

PS we currently have four workflow scripts ... 

![image](https://user-images.githubusercontent.com/464871/31833778-33720c14-b591-11e7-9e8a-8c9591501e96.png)

dlebauer committed:
More informative template

dlebauer committed:
merge develop into rmd_template

Merge remote-tracking branch 'origin' into rmd_template

# Conflicts:
#	CHANGELOG.md

tonygardella commented:
@dlebauer I'll bring this up next week in the weekly discussion. I ultimately think we just need a page in documentation outlining how to move to using pecan through Rstudio/R.

infotroph commented:
Without prejudice to the good points above about *whether* we want multiple workflows, a suggestion: If we *do* want this template, Would it make sense for it to live in `PEcAn.workflow` instead of `PEcAn.all`?

tonygardella commented:
@dlebauer @infotroph @mdietze I'm going to close this and open this as an issue for discussion. I like the idea of having a template to make it easy for people to run pecan outside of the web interface and perhaps direct people on where people can customize their runs. However, I don't like the idea of having to maintain multiple workflow scripts. 

I personally think we should have this live in documentation as a page describing and breaking down workflow.R so people can do a basic run from the command line or from within R studio.

dlebauer commented:
I am re-opening because I don't think that the underlying problem has been resolved.

infotroph commented:
@dlebauer OK to close this now that we have Rmd demos?

dlebauer commented:
It's different than a demo

infotroph commented:
Then please make changes to address the comments from eight years ago, all of which seem like reasonable requests to me.

dlebauer commented:
@infotroph ok will implement [Tony's suggestions](https://github.com/PecanProject/pecan/pull/1733#discussion_r146013898) to add:


```r
settings <- PEcAn.utils::do_conversions(settings)

settings <- PEcAn.DB::runModule.get.trait.data(settings)
PEcAn.settings::write.settings(settings, outputfile = 'pecan.TRAIT.xml')
```

And to be consistent with other demos will migrate to qmd.

There are other suggestions and ideas from long ago, but I am not sure if any necessary - this is distinct from the demo / tutorials, and is neither documentation nor a script.
